src/app/reporting/ReportScheduler.h

```diff
@@ -60,6 +79,29 @@ class ReportScheduler : public ReadHandler::Observer, public ICDStateObserver
         virtual Timestamp GetCurrentMonotonicTimestamp()   = 0;
     };
 
+    /**
+     * @class ReadHandlerNode
+     *
+     * @brief This class is in charge of determining when a ReadHandler is reportable depending on the monotonic timestamp of the
+     * system and the intervals of the ReadHandler. It inherits the TimerContext class to allow it to be used as a context for a
+     * TimerDelegate so the TimerDelegate can call the TimerFired method when the timer expires.
+     *
+     *  The Logic to determine if a ReadHandler is reportable at a precise timestamp is as follows:
+     *  1: The ReadHandler is in the CanStartReporting state
+     *  2: The minimal interval since last report has elapsed
+     *  3: The maximal interval since last report has elapsed or the ReadHandler is dirty
+     *  If the three conditions are met, the ReadHandler is reportable.
+     *
+     *  Additionnal flags have been provided for specific use cases:
+     *
+     *  CanbeSynced: Mechanism to allow the ReadHandler to emit a report if another readHandler is ReportableNow.
+     *  This flag can substitute the maximal interval condition or the dirty condition. It is currently only used by the
+     *  SynchronizedReportScheduler.
+     *
+     *  EngineRunScheduled: Mechanism to ensure that the reporting engine will see the ReadHandler as reportable if a timer fires.
```

Comment:
It means this: "This ReadHandler's timer fired, thus ScheduleReport was called, and we want the Report Engine to see the ReadHandler as reportable on the next run, regardless of what the clock thinks"

This could be renamed "TimerFired" flag if this seems more clear. Will update the description.

Reply:
How about ForceReportableOnNextEngineRun if that's what it means?